File: DataStructures/cpp/linked_list.cpp
#include <iostream>
using namespace std;

struct ListNode
{
    int val;
    ListNode *next;
    ListNode() : val(0), next(nullptr){};
    ListNode(int x) : val(x), next(nullptr){};
    ListNode(int x, ListNode *next) : val(x), next(next){};
};

int main()
{
    ListNode *x = new ListNode(0);
    ListNode *z = new ListNode(100);
    ListNode *y = new ListNode(10, z);

    x->next = y;
    while (x)
    {
        cout << x->val << "\t";
        x = x->next;
    }

    return 0;
}

File: DataStructures/python/linked_list.py
class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next


x = ListNode(0)
y = ListNode(1)
z = ListNode(12)
x.next = y
y.next = z


# or

x = ListNode(0, ListNode(1, ListNode(12)))

while(x):
    print(x.val)
    x = x.next
